src/odemis/gui/cont/acquisition.py

```diff
@@ -354,11 +355,19 @@ def __init__(self, tab_data, tab_panel):
         # Only possible to acquire if there are streams, and the chamber is
         # under vacuum
         tab_data.streams.subscribe(self.on_stream_chamber)
-        tab_data.main.chamberState.subscribe(self.on_stream_chamber, init=True)
+        tab_data.main.chamberState.subscribe(self.on_stream_chamber)
+        tab_data.roa.subscribe(self.on_stream_chamber, init=True)
 
         # Disable the "acquire image" button while preparation is in progress
         self._main_data_model.is_preparing.subscribe(self.on_preparation)
 
+    def _roa_isValid(self):
+        roa_valid = True
+        if hasattr(self._tab_data_model, "roa") and self._main_data_model.time_correlator is not None:
```

Comment:
In general, I'd rather avoid mix use of camelCase and under_scored function names. => `_roa_is_valid`.
I'd suggest to add a comment at the beginning of the function, as it might not be obvious why we check the ROA in a standard SECOM case:
```
Some streams (eg, TCSettingsStream) require a ROA for acquiring. So if any of this type of Stream is present, forbid to acquire until the ROA is defined.
```
As the comment implies, it's therefore not as simple as checking for `time_correlator`. It could be that there is a time correlator available, but the user doesn't want to use it (eg, just wants to use the confocal streams). So you need to check if `_tab_data_model.streams.value` contains any TCSettingsStream instead of looking for time_correlator.
That implies also that you've to also subscribe to `_tab_data_model.streams`.